src/components/organisms/Room/Participant.tsx

```diff
@@ -97,19 +102,22 @@ const Participant: React.FC<React.PropsWithChildren<ParticipantProps>> = ({
     }
   }, [audioTracks]);
 
-  useEffect(() => {
-    if (muted) {
-      const audioTrack = audioTracks[0];
-      if (audioTrack) {
-        audioTrack.detach();
+  const changeAudioState = useCallback(
+    (enable: boolean) => {
+      if (!enable) {
+        const audioTrack = audioTracks[0];
+        if (audioTrack) {
+          audioTrack.detach();
+        }
+      } else {
+        const audioTrack = audioTracks[0];
+        if (audioTrack && audioRef.current) {
+          audioTrack.attach(audioRef.current);
+        }
       }
-    } else {
-      const audioTrack = audioTracks[0];
-      if (audioTrack && audioRef.current) {
-        audioTrack.attach(audioRef.current);
-      }
-    }
-  }, [participant, muted, audioTracks]);
+    },
+    [audioTracks]
+  );
```

Comment:
Fail fast and reduced nesting of `if`s

```ts
if( !enable ) {
  return void audioTracks[0]?.detach();  // void operator optional, 2 lines can also serve
}
```

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/2221/commits/06234506347c70293948d6367762654d711fcc4b